src/jtag3.c

```diff
@@ -1597,6 +1600,27 @@ static int jtag3_parseextparms(const PROGRAMMER *pgm, const LISTID extparms) {
       }
     }
 
+    else if (str_starts(extended_param, "mode=") &&
```

Comment:
There should be an error message if -xmode=... is used but the programmer is neither snap nor pickit4.

Reply:
It is! See the next line.

$ ./avrdude -catmelice -patmega1284p -xmode=avr
avrdude error: invalid extended parameter 'mode=avr'
avrdude error: unable to parse extended parameter list

``` c
    else if (str_starts(extended_param, "mode=") &&
      (str_starts(pgmid, "pickit4") || str_starts(pgmid, "snap"))) {
```

Reply:
OK, I see: `-xmode=...` is only accepted for snap and pickit4, indeed.